### DISCUSSION TITLE
How to delete a Custom Button

### DISCUSSION THREAD

**Discussion by Guylby** [osmandapp/OsmAnd#19592] [Q&A] [answered]
Hi, 

I've been playing with the Custom Buttons but I can't seem to find a way to either rename or delete a custom created one. I can deactivate it, delete actions connected to it but can't delete or rename the button itselft from the menu.

Is there a hidden way somewhere?

**Comment by pebogufi** [comment]
Open
Configure profile / Configure screen / Buttons- Custom buttons / Quick action 
There you can edit, delete...

**Reply by Guylby** [reply]
Yes you can edit and delete action but there don't seem to be a way to rename or delete the button itself:

I want to delete or rename the "Test" custom button I created, but there don't seem to be an option here. The "Bin" button can only delete added actions to the custom button but not the button itself

| Customs button menu  | "Test" Custom button menu |
| ------------- | ------------- |
| ![438271720_734625832165175_858194190946099334_n](https://github.com/osmandapp/OsmAnd/assets/151167525/a74ce47a-0c3e-4bd9-aaef-175132604a2c) | ![438267625_442581728135500_6478985560223507867_n](https://github.com/osmandapp/OsmAnd/assets/151167525/e3feae5b-05a5-4702-832c-0d58f9bbb67b)|

**Reply by DmitryAlexei** [reply, marked as answer]
Quick actions are under development. See future task https://github.com/osmandapp/OsmAnd/issues/19258
As a workaround, you can delete custom quick action buttons by copying from another profile without custom quick actions.

**Reply by Guylby** [reply]
Thanks for the help!

**Comment by mariush444** [comment]
so it is realised but under development?
https://osmand.net/blog/osmand-android-4-7-released/#custom-buttons

**Reply by DmitryAlexei** [reply]
Yes, first iteration
they will be improved in future

**Comment by Sonwon1** [comment]
Why would you release a beta feature in a production app? Beta features belong in beta apps not production. This would never pass in my experience with software development!

**Reply by vshcherb** [reply]
There is an open issue to allow to remove it. It was overlooked. 
Possible workaround: to export/import settings and reset settings

P.S. emotional rhetorical questions don't make any sense and won't be replied. If you want to communicate with developers, please take that into account.

**Reply by Sonwon1** [reply]
My apologies if I sounded emotional. I just did not believe this should have been included in the release. Thank you for the explanation.

**Reply by vshcherb** [reply]
Release goes every 3 months and the issue that it popped to our attention quite late. We urgently picking up such issues for 1-2 weeks and provide at least workaround at max fix that issue https://github.com/osmandapp/OsmAnd/issues/19596

Please take into account that release is needed for many other issues that couldn't wait like OAuth for OSM, Google Play policies, etc

**Comment by 532910** [comment]
Is any update there?

**Reply by dmpr0** [reply]
Hi! Delete action is available in Menu.
<img width="300" height="auto" alt="share_5789187503674786905" src="https://github.com/user-attachments/assets/4d373072-13f5-4692-af7e-4a171c80f366" />

**Reply by 532910** [reply]
I can't see it, is this a known bug?

<img src="https://github.com/user-attachments/assets/a0e4b433-2e30-4e0a-8639-52db54dd29f4"  width="300" height="auto"/>

**Reply by dmpr0** [reply]
Back button doesn't appear too? In your screenshot, the AppBar is hidden. What Android version do you have?

**Reply by 532910** [reply]
Yes, with back button, it's not hidden completely, you can see it, and I even can press a button on it. Android 13, osmand 5.2.13 from fdroid

**Reply by 532910** [reply]
https://github.com/user-attachments/assets/d8b7fbe3-733b-42ad-ac49-bc5fa3bbbd4b

**Reply by dmpr0** [reply]
Thanks for sharing. Since I cannot quickly reproduce this on a device running Android 13, it would be best to create a separate issue.
